src/lib/KeyStore.js

```diff
@@ -62,16 +74,23 @@ class KeyStore {
             return null;
         }
 
-        if (!keyRecord.encrypted) {
-            return new Key(keyRecord.secret, keyRecord.type);
+        if (!KeyStore.isEncrypted(keyRecord)) {
+            // TODO: Compare stored type with purposeID to make sure?
```

Comment:
If someone was to use an Imagewallet from a different source, hence different `PURPOSE_ID`, what would be the expected behaviour of our code? Treat it as Entropy?

Reply:
Yes, just like in the core, we would treat it as entropy. That's why I reversed the checks, to handle PrivateKey specially, but in all other cases just make an Entropy from it.